### PR TITLE
upload_pack_by_branch: Replace deepen with deepen_level

### PR DESCRIPTION
**Pull request by mariusraht2**
Related to #3040

We need a new method upload_pack_by_commit which shares ~90% of it's source code with upload_pack_by_branch but needs a specifiable deepen_level import parameter. So this PR makes sure we can merge redundant and duplicate source code for these two methods into a new upload_pack method (will be introduced after merging this).

### CODE REVIEW

File: src/git/zcl_abapgit_git_transport.clas.abap [hunk 1]
```diff
@@ -4,18 +4,19 @@ CLASS zcl_abapgit_git_transport DEFINITION
   CREATE PUBLIC .
 
   PUBLIC SECTION.
+    TYPES: ty_deepen_level TYPE n LENGTH 2.
```
Comment (larshp) [inline src/git/zcl_abapgit_git_transport.clas.abap:7]: suggest being able to pass any number, there is no need to add restrictions(eg. passing 100 as deepen level wont work), or adding extra TYPES

**Comment by mbtools** [comment]
128 commits for this change? but looks like you got what you wanted in the end

**Comment by mariusraht2** [comment]
:D No, that are just the ones I merged from master into my related branch

**Comment by flaiker** [comment]
Then only submit the three changes? Lars will probably squash anyway but the process for updating your fork seems off.

**Comment by mariusraht2** [comment]
What would be the best way to update my fork? I do it by creating a PR for my master branch with the source larshp/master and then I merge it.

**Comment by flaiker** [comment]
Usually you don't create PRs on forks (unless they are hard forks living independently). I guess the PR creates a non fast forward merge commit which here shows up as new commit that should be merged.

I just do  
`git checkout master`  
`git fetch upstream`  
`git merge upstream/master master`  
`git push origin master`.

If you already have a feature branch you want to update I personally do a rebase but merging should work as well.

https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork

**Comment by g-back** [comment]
Alterantively you can continue using the `pull` app - I sent you a DM in slack where I explain what you can change

**Comment by mariusraht2** [comment]
Alright guys, thank you very much for the input! :) I'll change it accordingly in future PRs.

**Review by larshp — CHANGES_REQUESTED** [review]
suggest being able to pass any(reasnable) number, there is no need to add restrictions(eg. passing 10 as deepen level wont work with N length 1), or adding extra TYPES

also see https://github.com/larshp/abapGit/pull/3835#discussion_r483032946

**Comment by mariusraht2** [comment]
What do you mean? Declaration is with N DEFAULT 1 not LENGTH - do I miss something here?

**Comment by mariusraht2** [comment]
@larshp Should be fine now - replaced N with I. Forgot that N has a default length of 1 because it's a num-like character.

**Review by larshp — APPROVED** [review]

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**